contentcuration/contentcuration/static/js/edit_channel/models.js

```diff
@@ -261,119 +238,137 @@ var TopicTreeModelCollection = BaseCollection.extend({
 	list_name:"topictree-list"
 });
 
+var TagModel = BaseModel.extend({
+	root_list : "tag-list",
+	defaults: {
+		tag_name: "Untagged"
+    }
+   /* get_or_create:function(){
+		var collection = new TagCollection();
+		collection.get_or_create(this.get("tag_name"), this.get("tag_type"));
+	}*/
+});
+
+var TagCollection = BaseCollection.extend({
+	model: TagModel,
+	list_name:"tag-list"
+});
 
 /**** MODELS SPECIFIC TO FILE NODES ****/
 var FileModel = BaseModel.extend({
-	root_list:"file-list"
+	root_list:"file-list",
+	duplicate: function(contentnode){
```

Comment:
not sure if this is how file duplication is done when duplicate contentnode. I thought it was happening at the backend